src/catplus-common/src/models/bravo.rs

```diff
@@ -0,0 +1,234 @@
+            }
+        }
+
+        Ok(())
+    }
+}
+
+#[derive(Clone, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BravoAction {
+    pub action_name: ActionName,
+    pub start_time: String,
+    pub ending_time: String,
+    pub method_name: Option<String>,
+    pub equipment_name: String,
+    pub sub_equipment_name: Option<String>,
+    pub speed_shaker: Option<Observation>,
+    pub at_well: Option<BravoWell>,
+    pub dispense_state: Option<String>,
+    pub dispense_type: Option<String>,
+    pub has_sample: Option<BravoSample>,
+    pub temperature: Option<Observation>,
+    pub volume_evaporation_final: Option<Observation>,
+    pub has_solvent: Option<Solvent>,
+    #[serde(rename = "SPMEprocess")]
+    pub spme_process: Option<bool>,
+    pub has_cartridge: Option<Cartridge>,
+    pub start_duration: Option<Observation>,
+    pub ending_duration: Option<Observation>,
+    pub order: Option<String>,
+    pub product_identification: ProductIdentification,
+}
+
+impl InsertIntoGraph for BravoAction {
+    fn insert_into(&self, graph: &mut LightGraph, iri: SimpleTerm) -> anyhow::Result<()> {
+        for (pred, value) in [
+            (rdf::type_, &self.action_name.iri().as_simple() as &dyn InsertIntoGraph),
+            (allores::AFX_0000622, &(self.start_time.as_str() * xsd::dateTime).as_simple()),
+            (allores::AFR_0002423, &(self.ending_time.as_str() * xsd::dateTime).as_simple()),
+            (allores::AFR_0001606, &self.method_name.as_ref().clone().map(|s| s.as_simple())),
+            (allores::AFR_0001723, &self.equipment_name.as_simple()),
+            (cat::startDuration, &self.start_duration),
+            (cat::endingDuration, &self.ending_duration),
+            (
+                cat::subEquipmentName,
+                &self.sub_equipment_name.as_ref().clone().map(|s| s.as_simple()),
+            ),
+            (cat::speedInRPM, &self.speed_shaker),
+            (cat::volumeEvaporationFinal, &self.volume_evaporation_final),
+            (alloprop::AFX_0000060, &self.temperature),
+            (cat::hasSample, &self.has_sample),
+            (cat::hasSolvent, &self.has_solvent),
+            (cat::hasWell, &self.at_well),
+            (cat::preparesProduct, &self.product_identification),
+            (allores::AFR_0001164, &self.product_identification.peak_identifier.as_simple()),
+            (cat::hasCartridge, &self.has_cartridge),
+            (cat::order, &self.order.as_ref().clone().map(|s| s.as_simple())),
+            (alloqual::AFQ_0000111, &self.dispense_state.as_ref().clone().map(|s| s.as_simple())),
+            (cat::dispenseType, &self.dispense_type.as_ref().clone().map(|s| s.as_simple())),
+        ] {
+            value.attach_into(
+                graph,
+                Link { source_iri: iri.clone(), pred: pred.as_simple(), target_iri: None },
+            )?;
+        }
+        if let Some(spme_process) = self.spme_process {
```

Comment:
@cmdoret This i now mapped correctly, but feel free to improve the code and add it into the loop above.